Content.Server/GameObjects/Components/Damage/DestructibleComponent.cs

```diff
@@ -73,6 +73,10 @@ void IOnDamageBehavior.OnDamageThresholdPassed(object obj, DamageThresholdPassed
             {
                 destroyed = true;
                 var pos = Owner.Transform.GridPosition;
+                if(Owner.TryGetComponent<EntityStorageComponent>(out var entityStorage))
```

Comment:
I think you should just make `EntityStorageComponent` handle `IDestroyAct`.

Reply:
WIll do.